### ISSUE TITLE
WebSocket 2.1: Clarify that once the container has identified a MessageHandler for a message, the MessageHandler is used for the entirety of the message irrespective of any subsequent changes to the MessageHandlers configured for the Session.

### ISSUE THREAD

**Issue by pnicolucci** [OpenLiberty/open-liberty#20580]
Spec issues: 

1) `https://github.com/jakartaee/websocket/issues/190`
2) `https://github.com/jakartaee/websocket/issues/192`

See "A.1. Changes Between 2.1 and 2.0" here: https://github.com/jakartaee/websocket/blob/master/spec/src/main/asciidoc/WebSocket.adoc

**Comment by volosied** [comment]
I believe we are compliant here. I inspected our implementation (via Eclipse & the debugger) on an app which removed a MessageHandler to see what happens when two simultaneous requests came in.  The requests for the each websocket session were handled sequentially.  This means requests use the same MessageHandler for the whole request. Only once the second request was read did we encounter the following error: ```[WARNING ] CWWKH0025E: Client sent a message to Endpoint com.example.ProgrammaticEchoEndpoint which does not have @OnMessage method or MessageHandler for this type of message defined.```

As quoted in issue 190, "Once you start a websocket message, you continue to use the same MessageHandler for the entire message".  

Although my testing only tested with removal of a message handler, I believe the principle here holds if the message handler is swapped out for another (as reported in the original issue). 

I tested the the app with Tomcat, and encountered the same behavior (but Tomcat didn't report a MessageHandler was not found) . 


Additionally, no messages are read until after the onOpen call occurs (issue 191). We're compliant here, too. 

I'm not sure if tests can be added here as these are very time dependent issues.
